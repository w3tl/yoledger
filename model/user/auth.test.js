jest.mock('crypto');
jest.mock('argon2');

const RealDate = Date;
function mockDate(isoDate) {
  global.Date = class extends RealDate {
    constructor() {
      super();
      return new RealDate(isoDate);
    }
  };
}

describe('auth', () => {
  beforeAll(() => {
    mockDate('2018-06-20');
  });

  afterAll(() => {
    global.Date = RealDate;
  });

  test('createCredentials', async () => {
    const { createCredentials } = require('./auth');
    const result = await createCredentials({ password: 'password' });
    expect(result).toMatchSnapshot();
  });
});
