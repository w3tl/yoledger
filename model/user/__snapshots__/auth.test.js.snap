// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth createCredentials 1`] = `
Object {
  "algorithm": "argon2",
  "createdAt": 2018-06-20T00:00:00.000Z,
  "hash": "12345678901234567890123456789012password",
  "salt": "12345678901234567890123456789012",
  "valid": true,
}
`;
